stock_inventory_revaluation/models/stock_inventory_revaluation.py

```diff
@@ -0,0 +1,321 @@
+                # Products with manual inventory valuation are ignored because they
+                # do not need to create journal entries.
+                if product.valuation != "real_time":
+                    continue
+                # `remaining_qty` is negative if the move is out and delivered proudcts
+                #  that were not
+                # in stock.
+                qty_out = self._get_qty_out(line, remaining_qty)
+                move_vals["line_ids"] += line._create_accounting_entries(
+                    move, line, qty_out
+                )
+
+            # batch standard price computation avoid recompute quantity_svl at
+            # each iteration
+            products = self.env["product.product"].browse(
+                p.id for p in revaluation_to_add_byproduct.keys()
+            )
+            for (
+                product
+            ) in products:  # iterate on recordset to prefetch efficiently quantity_svl
+                if not float_is_zero(
+                    product.quantity_svl, precision_rounding=product.uom_id.rounding
+                ):
+                    product.with_company(revaluation.company_id).sudo().with_context(
+                        disable_auto_svl=True
+                    ).standard_price += (
+                        revaluation_to_add_byproduct[product] / product.quantity_svl
+                    )
+
+            move_vals["stock_valuation_layer_ids"] = [(6, None, valuation_layer_ids)]
+            # We will only create the accounting entry when there are defined lines
+            # (the lines will be those linked to products of real_time valuation
+            # category).
+            revaluation_vals = {"state": "done"}
+            if move_vals.get("line_ids"):
+                move = move.create(move_vals)
+                revaluation_vals.update({"account_move_id": move.id})
+            revaluation.write(revaluation_vals)
+            if revaluation.account_move_id:
+                move._post()
+
+            if (
+                revaluation.vendor_bill_id
+                and revaluation.vendor_bill_id.state == "posted"
+                and revaluation.company_id.anglo_saxon_accounting
+            ):
+                all_amls = (
+                    revaluation.vendor_bill_id.line_ids
+                    | revaluation.account_move_id.line_ids
+                )
+                for product in revaluation.stock_inventory_revaluation_line_ids.mapped(
+                    "product_id"
+                ):
+                    accounts = product.product_tmpl_id.get_product_accounts()
+                    input_account = accounts["stock_input"]
+                    all_amls.filtered(
+                        lambda aml: aml.account_id == input_account
+                        and aml.product_id == product
+                        and not aml.full_reconcile_id
+                    ).reconcile()
```

Comment:
Maybe we should check if the account is set to be reconciled? Otherwise you wil have an error while validating:

![image](https://user-images.githubusercontent.com/23449160/159229291-c32c3e63-cc55-4baa-aeeb-24e091d681d3.png)

Reply:
Good point! @LoisRForgeFlow , Added the logic to work with partial invoices. Also include a test for that. Thanks